tests/extension/oneapi_composite_device/enumerating_composite_devices.cpp

```diff
@@ -0,0 +1,157 @@
+
+  STATIC_REQUIRE(
+      std::is_same_v<std::vector<sycl::device>,
+                     decltype(platform.ext_oneapi_get_composite_devices())>);
+
+#endif
+}
+
+TEST_CASE("Test for free function get_composite_device",
+          "[oneapi_composite_device]") {
+#ifndef SYCL_EXT_ONEAPI_COMPOSITE_DEVICE
+  SKIP(
+      "The sycl_ext_oneapi_composite device extension is not supported by an "
+      "imlementation");
+#else
+
+  std::vector<sycl::device> composite_devices;
+
+  // get_composite_device may not throw
+  try {
+    composite_devices =
+        sycl::ext::oneapi::experimental::get_composite_devices();
+  } catch (sycl::exception& e) {
+    FAIL("get_composite_devices threw an exception: " << e.what());
+  }
+
+  {
+    INFO("Subsequent calls to get_composite_devices()");
+    std::vector<sycl::device> composite_devices2;
+    // get_composite_device may not throw
+    try {
+      composite_devices2 =
+          sycl::ext::oneapi::experimental::get_composite_devices();
+      INFO("Checking that subsequent call returns the same list of devices");
+      REQUIRE(composite_devices == composite_devices2);
+    } catch (sycl::exception& e) {
+      FAIL("get_composite_devices threw an exception: " << e.what());
+    }
+  }
+
+#endif
+}
+
+TEST_CASE("Test for platform::ext_oneapi_get_composite_devices",
+          "[oneapi_composite_device]") {
+#ifndef SYCL_EXT_ONEAPI_COMPOSITE_DEVICE
+  SKIP(
+      "The sycl_ext_oneapi_composite device extension is not supported by an "
+      "imlementation");
+#else
+
+  auto platform = sycl_cts::util::get_cts_object::platform();
+  std::vector<sycl::device> composite_devices;
+
+  // get_composite_device may not throw
+  try {
+    composite_devices = platform.ext_oneapi_get_composite_devices();
+  } catch (sycl::exception& e) {
+    FAIL("platform::ext_oneapi_get_composite_devices threw an exception: "
+         << e.what());
+  }
+
+  {
+    INFO("Subsequent calls to platform::ext_oneapi_get_composite_devices()");
+    std::vector<sycl::device> composite_devices2;
+    // get_composite_device may not throw
+    try {
+      composite_devices2 = platform.ext_oneapi_get_composite_devices();
+      INFO("Checking that subsequent call returns the same list of devices");
+      REQUIRE(composite_devices == composite_devices2);
+    } catch (sycl::exception& e) {
+      FAIL("platform::ext_oneapi_get_composite_devices threw an exception: "
+           << e.what());
+    }
+  }
+
+#endif
+}
+
+TEST_CASE(
+    "Test to check that composite devices are not returned as root devices",
+    "[oneapi_composite_device]") {
+#ifndef SYCL_EXT_ONEAPI_COMPOSITE_DEVICE
+  SKIP(
+      "The sycl_ext_oneapi_composite device extension is not supported by an "
+      "imlementation");
+#else
+
+  auto platform = sycl_cts::util::get_cts_object::platform();
+
+  std::vector<sycl::device> root_devices = platform.get_devices();
+  for (auto device : sycl::device::get_devices()) {
+    root_devices.push_back(device);
+  }
```

Comment:
Why do we need this? I thought `sycl::device::get_devices()` should already return the devices returned by `platform.get_devices()`, am I wrong?

Reply:
I don't understand what you mean, sorry. The highlighted piece of code firstly gets all the devices of the platform, and subsequently it adds all the devices returned by `sycl::device::get_devices()` to the same vector. What I mean is that I think there will be duplicates here. If we do add all the devices from `sycl::device::get_devices()`, why do we need to start with `platform.get_devices()`? Like I said, I would expect duplicates here.

Reply:
Apology, I misread the question and mistakenly though that it was asked about a different test.

I agree that `platform.get_devices()` should contain everything that `device::get_devices()` returns, so we can probably simplify the code here

Reply:
> I agree that platform.get_devices() should contain everything that device::get_devices() returns, so we can probably simplify the code here

I think it's actually the other way around, isn't it? The result of `device::get_devices()` should contain the result of `platform.get_devices()`.

Reply:
> I think it's actually the other way around, isn't it? The result of `device::get_devices()` should contain the result of `platform.get_devices()`.

Right :)

Reply:
Applied in f8b1ca288cbb9f405080482cf9b73cd3ed38b2eb